olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/TileRequest.h

```diff
@@ -62,8 +62,8 @@ class DATASERVICE_READ_API TileRequest final {
    *
    * @return A reference to the updated `TileRequest` instance.
    */
-  inline TileRequest& WithBillingTag(const boost::optional<std::string>& tag) {
-    billing_tag_ = tag;
+  inline TileRequest& WithBillingTag(const boost::optional<std::string> tag) {
```

Comment:
Moving from const only works if you have the according move constructor:
```cpp
boost::optional(const boost::optional&& rhs);
```
which I doubt it has and even if this only copies instead of moving.